runtime/onert/core/src/exec/DynamicShapeInference.cc

```diff
@@ -970,15 +970,26 @@ void DynamicShapeInferer::visit(const ir::operation::Transpose &op)
 
   // from op, access the buffer of second input to read new shape
   auto input_ind = op.getInputs().at(ir::operation::Transpose::Input::INPUT);
-  auto input_tensor = _tensor_registry->getITensor(input_ind);
-  auto input_shape = input_tensor->getShape();
+  auto input = _tensor_registry->getITensor(input_ind);
+  auto input_shape = input->getShape();
 
-  if (!input_tensor->is_dynamic())
+  if ((!input->is_dynamic()) && (!output->is_dynamic()))
```

Comment:
Not sure about this check. @wateret PTAL

Reply:
Okay, I will add some comment.